raiden-ts/src/path/epics.ts

```diff
@@ -254,9 +251,12 @@ export const pathFindServiceEpic = (
                   mergeMap(pfs =>
                     pfs.price.isZero()
                       ? of({ pfs, iou: undefined })
-                      : prepareNextIOU$(state, pfs, deps, httpTimeout, tokenNetwork).pipe(
-                          map(iou => ({ pfs, iou })),
-                        ),
+                      : prepareNextIOU$(
+                          pfs,
+                          tokenNetwork,
+                          cached$.pipe(pluck(0)) /* cached state$ */,
```

Comment:
I am mostly curious about the reasoning behind this change also adding everything on the state. was the problem that passing the state the other way would result in the state being old/out of date?

Reply:
It wasn't on my tests because it was quick, but could be. Notice we use it mostly to get the blockNumber, and it's fetched after signing and requesting lastIOU, which can take a long time, so in the end, the blockNumber could be outdated. This change ensures reactive state is always kept up to date.